plugins/woocommerce/includes/emails/class-wc-email.php

```diff
@@ -256,6 +256,27 @@ public function __construct() {
 
 		add_action( 'phpmailer_init', array( $this, 'handle_multipart' ) );
 		add_action( 'woocommerce_update_options_email_' . $this->id, array( $this, 'process_admin_options' ) );
+		add_action( 'woocommerce_email_footer', array( $this, 'mobile_messaging' ), 9 ); // Run before the default email footer.
+	}
+
+	/**
+	 * Add mobile messaging.
+	 */
+	public function mobile_messaging() {
+		if ( ! $this instanceof WC_Email_New_Order ) {
```

Comment:
Curious, why not move this method (and the matching `add_action()` call) to `WC_Email_New_Order`? 

Just thinking that this code—which is specific to that other class and is not applicable to all emails—would then be grouped with other code also specific to new order emails.

Reply:
Initially, I wasn't sure if "New Order" was the only type of mail to which we wanted to add messages to. But as it's clear now, I'll move this method and `add_action` directly to `WC_Email_New_Order`, thanks!